segmentation/utils/spatial_analysis.py

```diff
@@ -107,3 +111,79 @@ def calculate_channel_spatial_enrichment(dist_matrix, marker_thresholds, all_pat
         # stats.append((z, muhat, sigmahat, p, h, adj_p, marker_titles))
         stats.loc[point, :, :] = stats_xr.values
     return values, stats
+
+
+def calculate_phenotype_spatial_enrichment(all_patient_data, dist_mat, points=None,
+                                           bootstrap_num=1000, dist_lim=100):
+    """Spatial enrichment analysis based on cell phenotypes to find significant interactions between different
+    cell types, looking for both positive and negative enrichment. Uses bootstrapping to permute cell labels randomly.
+
+    Args:
+        all_patient_data: data including points, cell labels, and
+            cell expression matrix for all markers
+        dist_mat: cells x cells matrix with the euclidian
+            distance between centers of corresponding cells
+        points: patient labels to include in analysis. If argument is none, default is all labels used
+        bootstrap_num: number of permutations for bootstrap. Default is 1000
+        dist_lim: cell proximity threshold. Default is 100
+
+    Returns:
+        values: a list with each element consisting of a tuple of
+            closenum and closenumrand for each point included in the analysis
+        stats: an Xarray with dimensions (points, stats, number of markers, number of markers) The included stats
+            variables are:
+            z, muhat, sigmahat, p, h, adj_p, and marker_titles for each point in the analysis"""
+
+    patient_idx = "SampleID"
```

Comment:
These are the names, not indices, so it's better not to have index in the variable name to avoid confusion